File: solutions/1365/1365-kir3i.cpp
class Solution {
public:
    vector<int> smallerNumbersThanCurrent(vector<int>& nums) {
        unordered_map<int, int> cnt;
        vector<int> t = nums;
        sort(t.begin(), t.end());
        for(int i=0; i<t.size(); i++)
            if(!cnt.count(t[i]))
                cnt[t[i]] = i;
        
        vector<int> ans;
        for(const int &x: nums)
            ans.push_back(cnt[x]);
        return ans;
    }
};

File: solutions/1365/1365-kir3i.java
class Solution {
    public int[] smallerNumbersThanCurrent(int[] nums) {
        HashMap<Integer, Integer> cnt = new HashMap<Integer, Integer>();
        int[] t = Arrays.copyOf(nums, nums.length);
        Arrays.sort(t);
        for(int i=0; i<t.length; i++)
            if(!cnt.containsKey(t[i]))
                cnt.put(t[i], i);
        
        int[] ans = new int[nums.length];
        for(int i=0; i<nums.length; i++)
            ans[i] = cnt.get(nums[i]);
        return ans;
    }
}
